test/LocalStorageTest.coffee

```diff
@@ -79,3 +79,41 @@ exports.LocalStorageTest =
     test.equals(local.getItem('junk'), null)
     local._deleteLocation()
     test.done()
+
+  testEvents: (test) ->
+    localStorage = new LocalStorage('./scratch5')
+    key = null; oldVal = null; newVal = null; url = null
+    handleEvent = (evnt) ->
+      key = evnt.key
+      oldVal = evnt.oldValue
+      newVal = evnt.newValue
+      url = evnt.url
+
+    localStorage.on('storage', handleEvent)
+    
+    localStorage.setItem('a', 'something')
+    test.equal(localStorage.getItem('a'), 'something')
+    test.equal(key, 'a')
+    test.equal(oldVal, null)
+    test.equal(newVal, 'something')
+    test.equal(url, '')
```

Comment:
You'll have to filter out the beginning parts of the url when you put the filename in here.